Applications/ApplicationsLib/TestDefinition.cpp

```diff
@@ -0,0 +1,252 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "TestDefinition.h"
+
+#include <cstdlib>
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+#include "BaseLib/FileTools.h"
+#ifdef USE_PETSC
+#include "MeshLib/IO/VtkIO/VtuInterface.h"  // For petsc file name conversion.
+#include <petsc.h>
+#endif
+
+namespace
+{
+/// Test if the given string is convertible to a valid double value, not a NaN.
+bool isConvertibleToDouble(std::string const& s)
+{
+    std::size_t pos = 0;
+    double value;
+    try
+    {
+        value = std::stod(s, &pos);
+    }
+    catch (...)
+    {
+        OGS_FATAL("The given string '%s' is not convertible to double.",
+                  s.c_str());
+    }
+    if (pos != s.size())
+    {
+        OGS_FATAL(
+            "Only %d characters were used for double conversion of string '%s'",
+            pos, s.c_str());
+    }
+
+    if (std::isnan(value))
```

Comment:
compilation error with g++: error: ‘isnan’ is not a member of ‘std’

Reply:
This has been fixed here https://github.com/ufz/ogs/commit/66eb63ae98e13993eaa42534f80edde2ac57c9b2

Reply:
@bilke Thanks.